### PR TITLE
Update nethermind to chiado reboot

### PR DESCRIPTION
**Pull request by tropicar**
Now nethermind supports chiado testnet officialy. We have added the snap synchronization too.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -13,14 +13,16 @@ services:
...
       NETHERMIND_SYNCCONFIG_FASTSYNC: "true"
       NETHERMIND_JSONRPCCONFIG_ENABLED: "true"
       NETHERMIND_JSONRPCCONFIG_HOST: 0.0.0.0
       NETHERMIND_INITCONFIG_WEBSOCKETSENABLED: "true"
-      NETHERMIND_NETWORKCONFIG_DISCOVERYPORT: 31309
-      NETHERMIND_NETWORKCONFIG_P2PPORT: 31309
-      NETHERMIND_JSONRPCCONFIG_ADDITIONALRPCURLS: "http://0.0.0.0:8551|http;ws|net;eth;subscribe;engine;web3;client"
-      LOG: INFO
+      NETHERMIND_NETWORKCONFIG_DISCOVERYPORT: 30404
+      NETHERMIND_NETWORKCONFIG_P2PPORT: 30404
```
Comment (alexpeterson91) [inline docker-compose.yml]:
```suggestion
      NETHERMIND_NETWORKCONFIG_P2PPORT: 31309
```

File: docker-compose.yml [hunk 2]
```diff
@@ -13,14 +13,16 @@ services:
...
+      NETHERMIND_CONFIG: chiado
       NETHERMIND_SYNCCONFIG_FASTSYNC: "true"
       NETHERMIND_JSONRPCCONFIG_ENABLED: "true"
       NETHERMIND_JSONRPCCONFIG_HOST: 0.0.0.0
       NETHERMIND_INITCONFIG_WEBSOCKETSENABLED: "true"
-      NETHERMIND_NETWORKCONFIG_DISCOVERYPORT: 31309
-      NETHERMIND_NETWORKCONFIG_P2PPORT: 31309
-      NETHERMIND_JSONRPCCONFIG_ADDITIONALRPCURLS: "http://0.0.0.0:8551|http;ws|net;eth;subscribe;engine;web3;client"
-      LOG: INFO
+      NETHERMIND_NETWORKCONFIG_DISCOVERYPORT: 30404
```
Comment (alexpeterson91) [inline docker-compose.yml]:
```suggestion
      NETHERMIND_NETWORKCONFIG_DISCOVERYPORT: 31309
```

File: docker-compose.yml [hunk 3]
```diff
@@ -29,9 +31,5 @@ services:
       nofile:
         soft: 1000000
         hard: 1000000
-    deploy:
-      resources:
-        limits:
-          memory: 16384M
```
Comment (alexpeterson91) [inline docker-compose.yml]: keep this. nethermind still needs it and it wont hurt this

**Review by alexpeterson91 — CHANGES_REQUESTED** [review]
ports need to be kept as is or updated to the new port as defined by the new chain ID, reverting to 30404 does not work!

File: docker-compose.yml (hunk 3 above)
Comment (tropicar) [reply docker-compose.yml]: Added

File: docker-compose.yml (hunk 2 above)
Comment (tropicar) [reply docker-compose.yml]: Changed

File: docker-compose.yml (hunk 1 above)
Comment (tropicar) [reply docker-compose.yml]: Changed

**Comment by tropicar** [comment]
> ports need to be kept as is or updated to the new port as defined by the new chain ID, reverting to 30404 does not work!
We must use 31309 ports because there are conflicts if we use 30404.

File: dappnode_package.json [hunk 4]
```diff
@@ -3,7 +3,7 @@
     "version": "1.0.0",
     "upstreamVersion": "1.13.6",
     "shortDescription": "Nethermind Chiado client",
-    "description": "Nethermind - The world’s fastest Ethereum .NET Client and P2P Data Marketplace for the decentralised future\nOur flagship .NET Core-based Ethereum client is all about performance and flexibility. We have built a lean, stable and feature-rich gateway to allow anyone, anywhere, to take part in the decentralised future.",
+    "description": "Nethermind is a execution client for running Chiado network, Nethermind is a high-performance, highly configurable full Ethereum protocol client built on .NET that runs on Linux, Windows, and macOS, and supports Clique, Aura, Ethash, and Proof-of-Stake consensus algorithms.",
```
Comment (alexpeterson91) [inline dappnode_package.json]:
```suggestion
    "description": "Chiado Nethermind is a execution layer (EL) client for running the Gnosis longterm testnet `Chiado`, Nethermind is a high-performance, highly configurable full Ethereum protocol client built on .NET that runs on Linux, Windows, and macOS, and supports Clique, Aura, Ethash, and Proof-of-Stake consensus algorithms.",
```

File: dappnode_package.json [hunk 5]
```diff
@@ -3,7 +3,7 @@
     "version": "1.0.0",
     "upstreamVersion": "1.13.6",
     "shortDescription": "Nethermind Chiado client",
-    "description": "Nethermind - The world’s fastest Ethereum .NET Client and P2P Data Marketplace for the decentralised future\nOur flagship .NET Core-based Ethereum client is all about performance and flexibility. We have built a lean, stable and feature-rich gateway to allow anyone, anywhere, to take part in the decentralised future.",
+    "description": "Chiado Nethermind is a execution layer (EL) client for running the Gnosis longterm testnet `Chiado`, Nethermind is a high-performance, highly configurable full Ethereum protocol client built on .NET that runs on Linux, Windows, and macOS, and supports Clique, Aura, Ethash, and Proof-of-Stake consensus algorithms.",
```
Comment (alexpeterson91) [inline dappnode_package.json]:
```suggestion
    "description": "Chiado Nethermind is an Execution Layer (EL) client for running the Gnosis longterm testnet `Chiado`, Nethermind is a high-performance, highly configurable full Ethereum protocol client built on .NET that runs on Linux, Windows, and macOS, and supports Clique, Aura, Ethash, and Proof-of-Stake consensus algorithms.",
```

**Review by alexpeterson91 — APPROVED** [review]
All good i mean the P2P ports can be changed to the proper ones not a placeholder because 10200 is the new chainID for Chiado so P2P port for this repo under new specs is `32304`
`200*10=2,000+30303=32303+1=32304`
But it's not necessary now since it works with the old ports we set for it

**Comment by alexpeterson91** [comment]
@tropicar https://github.com/dappnode/DAppNode/issues/457#issuecomment-1299478325 can you update the main comment with the fixed formatting and the proper ports for chiado now that 10200 is its chainID per the official config? and idk i guess we should probably just do the ports now since this is the time before publishing updates for the reboot and its planned to be a long term testnet

File: docker-compose.yml [hunk 6]
```diff
@@ -13,14 +13,16 @@ services:
       - "data:/data"
     environment:
       MODE: normal
+      NETHERMIND_CONFIG: chiado
       NETHERMIND_SYNCCONFIG_FASTSYNC: "true"
       NETHERMIND_JSONRPCCONFIG_ENABLED: "true"
       NETHERMIND_JSONRPCCONFIG_HOST: 0.0.0.0
       NETHERMIND_INITCONFIG_WEBSOCKETSENABLED: "true"
       NETHERMIND_NETWORKCONFIG_DISCOVERYPORT: 31309
```
Comment (alexpeterson91) [inline docker-compose.yml]:
```suggestion
      NETHERMIND_NETWORKCONFIG_DISCOVERYPORT: 32304
```

File: docker-compose.yml [hunk 7]
```diff
@@ -13,14 +13,16 @@ services:
...
       NETHERMIND_NETWORKCONFIG_P2PPORT: 31309
```
Comment (alexpeterson91) [inline docker-compose.yml]:
```suggestion
      NETHERMIND_NETWORKCONFIG_P2PPORT: 32304
```

File: docker-compose.yml [hunk 8]
```diff
@@ -13,14 +13,16 @@ services:
...
-      NETHERMIND_JSONRPCCONFIG_ADDITIONALRPCURLS: "http://0.0.0.0:8551|http;ws|net;eth;subscribe;engine;web3;client"
-      LOG: INFO
+      NETHERMIND_JSONRPCCONFIG_ENGINEHOST: 0.0.0.0
+      NETHERMIND_JSONRPCCONFIG_ENGINEPORT: 8551
+      NETHERMIND_HEALTHCHECKSCONFIG_UIENABLED: "true"
       EXTRA_OPTS: ""
     ports:
       - "31309:31309/udp"
```
Comment (alexpeterson91) [inline docker-compose.yml]:
```suggestion
      - "32034:32034/udp"
```

File: docker-compose.yml [hunk 9]
```diff
@@ -13,17 +13,19 @@ services:
...
+      NETHERMIND_NETWORKCONFIG_DISCOVERYPORT: 32304
+      NETHERMIND_NETWORKCONFIG_P2PPORT: 32304
+      NETHERMIND_JSONRPCCONFIG_ENGINEHOST: 0.0.0.0
+      NETHERMIND_JSONRPCCONFIG_ENGINEPORT: 8551
+      NETHERMIND_HEALTHCHECKSCONFIG_UIENABLED: "true"
       EXTRA_OPTS: ""
     ports:
-      - "31309:31309/udp"
+      - "32034:32034/udp"
       - "31309:31309/tcp"
```
Comment (alexpeterson91) [inline docker-compose.yml]:
```suggestion
      - "32304:32304/tcp"
```

**Review by alexpeterson91 — APPROVED** [review]
just went ahead and did the ports. we should for all now.  i calculated all already theyre in the above comment link

**Pull request merged**